File: structural/composite/Application.java
package jarai.patterns.gof.structural.composite;


public class Application {

    public static void main(string[] args) {

        var meinComputer = new Computer("IBM", 100);

        var meineMutterplatine = new Mutterplatine("Asus", 120);
        meineMutterplatine.hinzufuegen(new Prozessor(2400, 300));
        meineMutterplatine.hinzufuegen(new Ram(512, 250));

        meinComputer.hinzufuegen(meineMutterplatine);
        meinComputer.hinzufuegen(new Festplatte("C:", 100));
        meinComputer.hinzufuegen(new Festplatte("D:", 100));

        meinComputer.anzeigen();

    }
}

File: structural/composite/Application.cpp
#include "Computer.h"
#include "Mutterplatine.h"
#include "Prozessor.h"
#include "Ram.h"
#include "Festplatte.h"

int main() {

	Computer meinComputer("IBM", 100);
	Festplatte hd1("C:", 100);
	Festplatte hd2("D:", 100);

	Mutterplatine meineMutterplatine("Asus", 120);
	Prozessor prozessor = Prozessor(2400, 300);
	Ram ram = Ram(512, 250);
	meineMutterplatine.hinzufuegen( prozessor);
	meineMutterplatine.hinzufuegen( ram);

	meinComputer.hinzufuegen(meineMutterplatine);
	meinComputer.hinzufuegen(hd1);
	meinComputer.hinzufuegen(hd2 );

	meinComputer.anzeigen();

}

